src/org/ods/component/ScanWithSonarStage.groovy

```diff
@@ -1,29 +1,46 @@
 package org.ods.component
 
+import com.cloudbees.groovy.cps.NonCPS
+
+import org.ods.services.BitbucketService
 import org.ods.services.SonarQubeService
 
 class ScanWithSonarStage extends Stage {
   public final String STAGE_NAME = 'SonarQube Analysis'
+  private BitbucketService bitbucket
   private SonarQubeService sonarQube
 
-  ScanWithSonarStage(def script, IContext context, Map config, SonarQubeService sonarQube) {
+  ScanWithSonarStage(def script, IContext context, Map config, BitbucketService bitbucket, SonarQubeService sonarQube) {
     super(script, context, config)
-    if (!config.requireQualityGatePass) {
+    if (config.branch) {
+      config.eligibleBranches = config.branch.split(',')
+    } else {
+      config.eligibleBranches = context.sonarQubeBranch.split(',')
+    }
+    if (!config.containsKey('analyzePullRequests')) {
+      config.analyzePullRequests = true
+    }
+    if (!config.containsKey('requireQualityGatePass')) {
       config.requireQualityGatePass = false
```

Comment:
Can't you do 
`config.requireQualityGatePass = config.containsKey('requireQualityGatePass')`
here? Or do you think it is less readable?

Reply:
No, one can set `requireQualityGatePass` explicitly to `false`.